cmd/share-secret/main.go

```diff
@@ -59,12 +64,30 @@ func PostSecret() http.HandlerFunc {
 	}
 }
 
-func GetSecret(w http.ResponseWriter, r *http.Request) {
+func GetSecret() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		key := mux.Vars(r)["key"]
+		if key == "" {
+			http.Error(w, "The path is not complete", http.StatusInternalServerError)
+			return
+		}
+
+		data := mData[key]
+		if data.Secret == "" {
```

Comment:
We should not check this here, but when we recieve the key in the `PostSecret()` endpoint